File: CroquetaRecetaBuilder.java
public class CroquetaRecetaBuilder extends RecetaBuilder{
    public CroquetaRecetaBuilder() {
        super.receta = new Receta();
    }

    @Override
    public void BuildIngredientes() {
        receta.setIngredientes("Harina, leche, huevo, jamón, sal, aceite");
    }

    @Override
    public void BuildPasos() {
        receta.setPasos("Mezclar la harina, la leche y el huevo para hacer la masa. Añadir el jamón y la sal a la masa. Freír la masa en aceite caliente hasta que estén doradas.");
    }

    @Override
    public void BuildTiempo() {
        receta.setTiempoPreparacion(45);
    }
}

File: practica4/recetario_flutter/lib/croquetarecetabuilder.dart
import 'receta.dart';
import 'recetabuilder.dart';

class CroquetaRecetaBuilder extends RecetaBuilder {
  CroquetaRecetaBuilder() {
    super.receta = Receta('Croquetas');
  }

  @override
  void buildIngredientes() {
    receta.setIngredientes("Leche, harina, mantequilla, jamón, huevo, pan rallado, aceite");
  }

  @override
  void buildPasos() {
    receta.setPasos("Hacer una bechamel con la leche, la harina y la mantequilla. Añadir el jamón y dejar enfriar. Formar las croquetas, pasar por huevo y pan rallado y freír en aceite caliente.");
  }

  @override
  void buildTiempo() {
    receta.setTiempoPreparacion(60);
  }
  
}